File: admin_flutter/lib/features/settings/presentation/screens/settings_screen.dart
/**
 * TigerWallet Admin - Settings Screen
 */

import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:go_router/go_router.dart';
import '../../../../core/theme/app_theme.dart';
import '../../../../core/constants/app_constants.dart';

class SettingsScreen extends ConsumerWidget {
  const SettingsScreen({super.key});

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final isDark = Theme.of(context).brightness == Brightness.dark;
    return Scaffold(
      appBar: AppBar(title: const Text('Settings')),
      body: ListView(
        padding: const EdgeInsets.all(16),
        children: [
          _buildSection(context, 'Account', [
            _buildSettingTile(context, Icons.person, 'Profile', 'Manage your profile', () => context.go('${AppConstants.settingsRoute}/profile'), isDark),
            _buildSettingTile(context, Icons.lock, 'Security', 'Password and 2FA', () {}, isDark),
            _buildSettingTile(context, Icons.notifications, 'Notifications', 'Push and email notifications', () {}, isDark),
          ], isDark),
          const SizedBox(height: 16),
          _buildSection(context, 'Appearance', [
            _buildSettingTile(context, Icons.dark_mode, 'Theme', 'Dark/Light mode', () {}, isDark),
            _buildSettingTile(context, Icons.language, 'Language', 'English, Spanish, etc.', () {}, isDark),
          ], isDark),
          const SizedBox(height: 16),
          _buildSection(context, 'System', [
            _buildSettingTile(context, Icons.storage, 'Backup', 'Database backups', () {}, isDark),
            _buildSettingTile(context, Icons.webhook, 'Webhooks', 'Manage webhooks', () {}, isDark),
            _buildSettingTile(context, Icons.api, 'API Keys', 'Manage API keys', () {}, isDark),
          ], isDark),
          const SizedBox(height: 16),
          _buildSection(context, 'Support', [
            _buildSettingTile(context, Icons.help, 'Help Center', 'Documentation and guides', () {}, isDark),
            _buildSettingTile(context, Icons.info, 'About', 'Version 1.0.0', () {}, isDark),
          ], isDark),
          const SizedBox(height: 24),
          ElevatedButton(
            onPressed: () {},
            style: ElevatedButton.styleFrom(backgroundColor: AppTheme.errorColor),
            child: const Text('Sign Out'),
          ),
        ],
      ),
    );
  }

  Widget _buildSection(BuildContext context, String title, List<Widget> children, bool isDark) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(title, style: Theme.of(context).textTheme.titleMedium?.copyWith(fontWeight: FontWeight.bold, color: isDark ? Colors.white : const Color(0xFF1A1A2E))),
        const SizedBox(height: 12),
        Container(
          decoration: BoxDecoration(color: isDark ? AppTheme.darkCard : AppTheme.lightCard, borderRadius: BorderRadius.circular(16)),
          child: Column(children: children),
        ),
      ],
    );
  }

  Widget _buildSettingTile(BuildContext context, IconData icon, String title, String subtitle, VoidCallback onTap, bool isDark) {
    return ListTile(
      leading: Icon(icon, color: AppTheme.primaryColor),
      title: Text(title, style: const TextStyle(fontWeight: FontWeight.w600)),
      subtitle: Text(subtitle, style: Theme.of(context).textTheme.bodySmall),
      trailing: const Icon(Icons.chevron_right),
      onTap: onTap,
    );
  }
}

File: admin_android/app/src/main/java/com/tigerwallet/admin/ui/screens/settings/SettingsScreen.kt
package com.tigerwallet.admin.ui.screens.settings

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.unit.dp

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun SettingsScreen(onNavigateBack: () -> Unit, onNavigateToProfile: () -> Unit) {
    Scaffold(topBar = { TopAppBar(title = { Text("Settings") }, navigationIcon = { IconButton(onClick = onNavigateBack) { Icon(Icons.Default.ArrowBack, contentDescription = "Back") } }) }) { padding ->
        LazyColumn(modifier = Modifier.fillMaxSize().padding(padding), contentPadding = PaddingValues(16.dp)) {
            item { Text("Account", style = MaterialTheme.typography.titleMedium) }
            item { Card(modifier = Modifier.fillMaxWidth()) { Column { SettingsItem(Icons.Default.Person, "Profile", "Manage your profile", onNavigateToProfile); Divider(); SettingsItem(Icons.Default.Lock, "Security", "Password and 2FA", {}); Divider(); SettingsItem(Icons.Default.Notifications, "Notifications", "Push and email", {}) } } }
            item { Spacer(modifier = Modifier.height(16.dp)) }
            item { Text("Appearance", style = MaterialTheme.typography.titleMedium) }
            item { Card(modifier = Modifier.fillMaxWidth()) { Column { SettingsItem(Icons.Default.DarkMode, "Theme", "Dark/Light mode", {}); Divider(); SettingsItem(Icons.Default.Language, "Language", "English", {}) } } }
            item { Spacer(modifier = Modifier.height(16.dp)) }
            item { Text("System", style = MaterialTheme.typography.titleMedium) }
            item { Card(modifier = Modifier.fillMaxWidth()) { Column { SettingsItem(Icons.Default.Storage, "Backup", "Database backups", {}); Divider(); SettingsItem(Icons.Default.Webhook, "Webhooks", "Manage webhooks", {}); Divider(); SettingsItem(Icons.Default.Key, "API Keys", "Manage API keys", {}) } } }
            item { Spacer(modifier = Modifier.height(24.dp)) }
            item { Button(onClick = {}, modifier = Modifier.fillMaxWidth(), colors = ButtonDefaults.buttonColors(containerColor = MaterialTheme.colorScheme.error)) { Text("Sign Out") } }
        }
    }
}

@Composable
fun SettingsItem(icon: ImageVector, title: String, subtitle: String, onClick: () -> Unit) {
    ListItem(headlineContent = { Text(title) }, supportingContent = { Text(subtitle) }, leadingContent = { Icon(icon, contentDescription = null) }, modifier = Modifier.clickable(onClick = onClick), trailingContent = { Icon(Icons.Default.ChevronRight, contentDescription = null) })
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun ProfileScreen(onNavigateBack: () -> Unit) {
    Scaffold(topBar = { TopAppBar(title = { Text("Profile") }, navigationIcon = { IconButton(onClick = onNavigateBack) { Icon(Icons.Default.ArrowBack, contentDescription = "Back") } }) }) { padding ->
        Column(modifier = Modifier.fillMaxSize().padding(padding).padding(16.dp), horizontalAlignment = Alignment.CenterHorizontally) {
            Icon(Icons.Default.Person, contentDescription = null, modifier = Modifier.size(80.dp))
            Spacer(modifier = Modifier.height(16.dp))
            OutlinedTextField(value = "Admin User", onValueChange = {}, label = { Text("Name") }, modifier = Modifier.fillMaxWidth())
            Spacer(modifier = Modifier.height(12.dp))
            OutlinedTextField(value = "admin@tigerwallet.com", onValueChange = {}, label = { Text("Email") }, modifier = Modifier.fillMaxWidth())
            Spacer(modifier = Modifier.height(24.dp))
            Button(onClick = {}, modifier = Modifier.fillMaxWidth()) { Text("Save Changes") }
        }
    }
}
